### PR TITLE
Extend `algoliaadminkey` with additional checks

### PR DESCRIPTION
**Pull request by 0x2b3bfa0**
### Description
This pull request adds verification for other kinds of sensitive Algolia API keys apart from Admin keys, as per https://www.secjuice.com/api-misconfiguration-data-breach. 

This pull reques also enhances the detection of Algolia keys by looking also for Algolia DocSearch keywords and option names on the official libraries ([example](https://github.com/algolia/docsearch/blob/90f3c6aabbc324fe49e9a1dfe0906fcd4d90f27b/README.md?plain=1#L68-L73)).

### Checklist
* [x] Tests passing (`make test-community`)?
* [x] Lint passing (`make lint` this requires [golangci-lint](https://golangci-lint.run/welcome/install/#local-installation))?

### CODE REVIEW

**Comment by zricethezav** [comment]
@0x2b3bfa0 that blog is from 2020. Did you test this locally? Do you have a screenshot of the integration test passing with a valid credential?

**Comment by 0x2b3bfa0** [comment]
> _@0x2b3bfa0 that blog is from 2020._

Indeed, but still pretty relevant: an independent security researcher notified my company about this issue a couple days ago (we were affected), and that's precisely why I opened this pull request.

We use Trufflehog to scan all our repositories, and it didn't catch the overpermissioned tokens: firstly because of the missing DocSearch prefixes (#3458), and secondly because of this (#3459).

> _Did you test this locally?_

Yes, <sub>but thanks for asking, because the pull request was broken (missing imports) due to a bad Git merge. 🙈</sub> 

> _Do you have a screenshot of the integration test passing with a valid credential?_

Captura de pantalla 2024-10-19 a las 7 39 33

**Comment by 0x2b3bfa0** [comment]
@zricethezav, locally everything seems to work "fine" now; I wonder about continuous integration will agree. 🤞🏼

File: pkg/detectors/algoliaadminkey/algoliaadminkey.go [hunk 1]
```diff
@@ -22,14 +24,14 @@ var (
 	client = common.SaneHttpClient()
 
 	// Make sure that your group is surrounded in boundary characters such as below to reduce false positives.
-	keyPat = regexp.MustCompile(detectors.PrefixRegex([]string{"algolia"}) + `\b([a-zA-Z0-9]{32})\b`)
-	idPat  = regexp.MustCompile(detectors.PrefixRegex([]string{"algolia"}) + `\b([A-Z0-9]{10})\b`)
+	keyPat = regexp.MustCompile(detectors.PrefixRegex([]string{"algolia", "docsearch", "apiKey"}) + `\b([a-zA-Z0-9]{32})\b`)
+	idPat  = regexp.MustCompile(detectors.PrefixRegex([]string{"algolia", "docsearch", "appId"}) + `\b([A-Z0-9]{10})\b`)
```
Comment (0x2b3bfa0) [inline pkg/detectors/algoliaadminkey/algoliaadminkey.go:28]: @kashifkhan0771, regarding your comment from https://github.com/trufflesecurity/trufflehog/pull/3458#discussion_r1808314219:

> _I believe `appId` and `apiKey` is a very common term which can lead to false positives._

We're detecting these prefixes only in files that already contain the `algolia` or `docsearch` keyword, so I believe false positives would be very rare (and would be found through verification). Not including these terms might lead to false negatives, which —to my mind — is a bit more concerning.

**Comment by 0x2b3bfa0** [comment]
@zricethezav, I've consolidated #3458 into this pull request and made sure all tests pass.

**Review by zricethezav — APPROVED** [review]

**Pull request merged**